src/shared/codelens/codeLensUtils.ts

```diff
@@ -139,3 +147,36 @@ export function getMetricDatum({command, isDebug, runtime}: {
         }
     }
 }
+
+async function getAssociatedSamTemplate(
+    documentUri: vscode.Uri,
+    workspaceFolderUri: vscode.Uri
+): Promise<vscode.Uri> {
+
+    // Get Template files in Workspace
+    const templatesAsync: AsyncIterableIterator<vscode.Uri> = detectLocalTemplates({
+        workspaceUris: [workspaceFolderUri]
+    })
+
+    // See which templates (if any) are in a folder, or parent folder, of the document of interest
+    const templates = await toArrayAsync(templatesAsync)
+    const candidateTemplates = templates
+        .filter(template => {
+            const folder = dirname(template.fsPath)
+
+            return documentUri.fsPath.indexOf(folder) === 0
+        })
+
+    if (candidateTemplates.length === 0) {
+        throw new Error(
```

Comment:
This is an important regression. What prevents us from returning `undefined` instead?

Reply:
The behavior is consistent with what happens when [no workspace folder can be found](https://github.com/aws/aws-toolkit-vscode/pull/387/files/31c9042cae423788fa8f8ae5ea44f712a6cee926#diff-4a09b26abadcd1030a9d7e8172cc714fR52)